lib/backend_service/odyssey/odyssey_client.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:typed_data';
+import 'dart:async';
+
+/// Minimal abstraction over the Odyssey API used by providers. This allows
+/// swapping implementations (real HTTP client, mock, or unit-test doubles)
+/// while keeping providers free from direct dependency on ApiService.
+abstract class OdysseyClient {
```

Comment:
Given that a future option will be NanoDLP based, should we change this to `BackendClient`?